module_utils/azure_rm_common.py

```diff
@@ -310,6 +316,17 @@ def __init__(self, derived_arg_spec, bypass_checks=False, no_log=False,
         self.log("setting subscription_id")
         self.subscription_id = self.credentials['subscription_id']
 
+        # get authentication authority
+        # for adfs, user could pass in authority or not.
+        # for others, use default authority from cloud environment
+        if self.credentials.get('authority') is None:
+            self._authority = self._cloud_environment.endpoints.active_directory
```

Comment:
So the _authority is an endpoint or something else?

Reply:
yes, something like: https://login.windows.net, different cloud env has different authority, and it's configurable for adfs scenario.